cmd/system-probe/modules/eventmonitor_linux.go

```diff
@@ -10,15 +10,40 @@ package modules
 import (
 	"github.com/DataDog/datadog-agent/cmd/system-probe/api/module"
 	"github.com/DataDog/datadog-agent/cmd/system-probe/config"
+	sysconfigtypes "github.com/DataDog/datadog-agent/cmd/system-probe/config/types"
+	"github.com/DataDog/datadog-agent/comp/core/workloadmeta"
 	coreconfig "github.com/DataDog/datadog-agent/pkg/config"
+	netconfig "github.com/DataDog/datadog-agent/pkg/network/config"
+	procmon "github.com/DataDog/datadog-agent/pkg/process/monitor"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-agent/pkg/util/optional"
 )
 
 // EventMonitor - Event monitor Factory
 var EventMonitor = module.Factory{
 	Name:             config.EventMonitorModule,
 	ConfigNamespaces: eventMonitorModuleConfigNamespaces,
-	Fn:               createEventMonitorModule,
+	Fn:               createEventMonitorModuleLinux,
 	NeedsEBPF: func() bool {
 		return !coreconfig.SystemProbe.GetBool("runtime_security_config.ebpfless.enabled")
 	},
 }
+
+func createEventMonitorModuleLinux(config *sysconfigtypes.Config, wmeta optional.Option[workloadmeta.Component]) (module.Module, error) {
```

Comment:
I think it might be cleaner if you had a function that created the procmon monitor only (on windows that would just return nil) and then called it from `createEventMonitor`.